fw/cache.c

```diff
@@ -261,22 +261,26 @@ static DEFINE_PER_CPU(char[RESP_BUF_LEN], g_c_buf);
 static TfwStr g_crlf = { .data = S_CRLF, .len = SLEN(S_CRLF) };
 
 /* Iterate over request URI and Host header to process request key. */
-#define TFW_CACHE_REQ_KEYITER(c, uri_path, host, u_end, h_start, h_end)	\
+#define TFW_CACHE_REQ_KEYITER(c, uri_path, host, u_end, h_start,	\
+			      h_end, u_fin, h_fin)			\
+	u_fin = h_fin = false;						\
```

Comment:
Shouldn't we initialize the variables like:
```C
u_fin = TFW_STR_EMPTY(uri_path);
h_fin = TFW_STR_EMPTY(host);
```

The current contract for the macro is unclear: whether we rely on the fact that `host` is never empty or are we OK to process empty host?